src/app/reporting/ReportScheduler.h

```diff
@@ -89,7 +89,8 @@ class ReportScheduler : public ReadHandler::Observer, public ICDStateObserver
         bool IsReportableNow(const Timestamp & now) const
         {
             return (mReadHandler->CanStartReporting() &&
-                    (now >= mMinTimestamp && (mReadHandler->IsDirty() || now >= mMaxTimestamp || CanBeSynced())));
+                    ((now >= mMinTimestamp && (mReadHandler->IsDirty() || now >= mMaxTimestamp || CanBeSynced())) ||
+                     IsEngineRunScheduled()));
```

Comment:
This could use some documentation, explaining how if this specific handler has been scheduled to be reported already, we should treat it as reportable...

Reply:
Added note to the method to document the flag usage.